### PR TITLE
feat: [CI-4089]: Added animation and git authentication view enhanced for SaaS and Onprem setup

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
##### Summary:

1. Added animation to be shown during infrastructure provisioning.

https://user-images.githubusercontent.com/48940704/164887768-f6b3fa81-b264-4154-9039-de17501d2931.mov

2. Added on-prem view for git providers.

image

image

image

##### Jira Links:

https://harness.atlassian.net/browse/CI-4089

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Comment by mtran7** [comment]
It's hard to predict if the user has read the initial slides before the carousel moves... Quite a bit of text/animations.

Since they will be waiting for 2 minutes, should we have them click on the dots themselves? 
image

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 1]
```diff
@@ -176,25 +185,290 @@ const SelectGitProviderRef = (
...
 
+  //#region form view
+
   const permissionsForSelectedGitProvider = GitProviderPermissions.filter(
     (providerPermissions: GitProviderPermission) => providerPermissions.type === gitProvider?.type
   )[0]
 
+  const getButtonLabel = React.useCallback((): string => {
+    switch (gitProvider?.type) {
+      case 'Github':
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:196]: Can refer to an enum const for gitprovider types

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 2]
```diff
@@ -176,25 +185,290 @@ const SelectGitProviderRef = (
...
+  //#region methods exposed via ref
+
+  const markFieldsTouchedToShowValidationErrors = React.useCallback((): void => {
+    const { values, setFieldTouched } = formikRef.current || {}
+    const { accessToken, accessKey, applicationPassword, username, url } = values || {}
+    if (!authMethod) {
+      setFieldTouched?.('gitAuthenticationMethod', true)
+      return
+    }
+    if (gitProvider?.type === 'Github') {
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:315]: Refer to a const

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx]: Done, moved all connector references to constants.

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Review by mtran7 — APPROVED** [review]

**Review by nanaxuHarness — APPROVED** [review]

**Pull request merged**